### ISSUE TITLE
Gitea: set `update_at` field when updating issues

### ISSUE THREAD

**Issue by viceice** [renovatebot/renovate#26868]
### Describe the proposed change(s).

We should set `update_at` field when updating issues, otherwise the edits won't show up in history.

![image](https://github.com/renovatebot/renovate/assets/1798109/0cdae39c-32f2-469d-a127-3e0aae836999)

**Comment by RahulGautamSingh** [comment]
It doesn't seem like there is a `update_at` field in the request schema 
https://docs.gitea.com/api/1.20/#tag/issue/operation/issueEditIssue

**Comment by viceice** [comment]
interesting
https://gitea.com/api/swagger#/issue/issueEditIssue

forgejo does have it. 🤔

https://codeberg.org/api/swagger#/issue/issueEditIssue

**Comment by viceice** [comment]
can you check what gitea does if we set it? the API docs are generated manually from a swagger file which is sometimes out of sync with the real API.

**Comment by RahulGautamSingh** [comment]
We are allowed to set it in the request object but it has no effect.

**Comment by viceice** [comment]
ok, we should set it, it has an effect on Forgejo

https://github.com/go-gitea/gitea/blob/fc1bae00a4cc07d832eb9405ec8fd8f4e52c0197/modules/structs/issue.go#L101

https://codeberg.org/forgejo/forgejo/src/commit/8c0e5d07f8a95ebe4e6102591302effbe13f58e3/modules/structs/issue.go#L101

https://codeberg.org/forgejo/forgejo/commit/bd47ee33c20e53ae616a7e53d63c3b51809585fb

**Comment by RahulGautamSingh** [comment]
I tried implementing this by adding a simple `updated_at` field to the `gitea.ensureIssue` fn. As per the codeberg and forgejo api docs, the `updated_at` field should be a datetime string. Hence I used the following code:
```
update_at: DateTime.utc().toISO();
```

But this failed with a 403 error, with the message : 
```
"response": {
           "statusCode": 403,
           "statusMessage": "Forbidden",
           "body": {
             "message": "unallowed update date",
             "url": "https://codeberg.org/api/swagger"
           },
```

I even tried hardocoding the datetime string to `2024-02-14T04:06:14.438Z` which I am sure should work because I tried it out in the codeswagger and forgejo api playground. But I got the same error.

**Comment by viceice** [comment]
it seems there also a bug in Forgejo 

https://codeberg.org/Codeberg/forgejo/src/commit/40d2791343cff4c4f79be1101fa2a1d4b87d25fc/services/issue/issue.go#L326

I think this should be after the condition, so it gets an auto update from DB?
will ask on Forgejo dev matrix channel

**Comment by RahulGautamSingh** [comment]
Yeah, the first code shuold work by itself. Since the check is to prevent setting udpate time before issue creation date and later than now.

**Comment by viceice** [comment]
-https://codeberg.org/forgejo/forgejo/src/commit/2da33aae2d17ae5afb1908ef1caeacdfe5e7580e/models/issues/issue_update.go#L253
- https://codeberg.org/forgejo/forgejo/src/commit/2da33aae2d17ae5afb1908ef1caeacdfe5e7580e/routers/api/v1/repo/issue.go#L819-L821

Only the UI is causing a history entry. so setting `updated_at` isn't enough 😢

**Comment by RahulGautamSingh** [comment]
> Only the UI is causing a history entry. so setting `updated_at` isn't enough 😢

I had my doubts regarding this from the beginning 🕵️. Also, as a design choice I would prefer updated_at field to be added by default, and only be removed if user opts for it

**Comment by viceice** [comment]
we don't need it, it will automatically updated, but no history is written, that's why we don't see it on ui.

**Comment by viceice** [comment]
closing, it's a Forgejo / Gitea issue
